packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -28,6 +29,30 @@ export default function assets({
 
 	globalThis.astroAsset = {};
 
+	const UNSUPPORTED_ADAPTERS = new Set([
```

Comment:
It's okay for the error to be here, as this Vite plugin is only loaded when `experimental.assets` is enabled. 

It's currently unclear if we'll add non-Node support before 3.0 (mostly depends on deps), but if we don't, we'll have to move this warning somewhere else where we have access to the Astro config, maybe in the `astro:assets` virtual module